tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java

```diff
@@ -0,0 +1,270 @@
+                        createDepProposal( context, proposals, bundle );
+                    }
+                }
+            }
+        }
+        catch( Exception e )
+        {
+            GradleCore.logError( "Gradle dependence got error", e );
+        }
+    }
+
+    private void importNotFoundProposal(
+        IInvocationContext context, IProblemLocation problem, Collection<IJavaCompletionProposal> proposals )
+    {
+        ASTNode selectedNode = problem.getCoveringNode( context.getASTRoot() );
+
+        if( selectedNode == null )
+        {
+            return;
+        }
+
+        ImportDeclaration importDeclaration =
+            (ImportDeclaration) ASTNodes.getParent( selectedNode, ASTNode.IMPORT_DECLARATION );
+
+        if( importDeclaration == null )
+        {
+            return;
+        }
+
+        String importName = importDeclaration.getName().toString();
+        List<String> serviceWrapperList;
+        List<String> servicesList;
+        boolean depWrapperCanFixed = false;
+
+        try
+        {
+            serviceWrapperList = TargetPlatformUtil.getServiceWrapperList().getServiceList();
+            servicesList = TargetPlatformUtil.getServicesList().getServiceList();
+
+            if( serviceWrapperList.contains( importName ) )
+            {
+                ServiceContainer bundle = TargetPlatformUtil.getServiceWrapperBundle( importName );
+                createDepProposal( context, proposals, bundle );
+            }
+
+            if( !depWrapperCanFixed )
+            {
+                if( servicesList.contains( importName ) )
+                {
+                    ServiceContainer bundle = TargetPlatformUtil.getServiceBundle( importName );
+                    createDepProposal( context, proposals, bundle );
+                }
+            }
+        }
+        catch( Exception e )
+        {
+            GradleCore.logError( "Gradle dependence got error", e );
+        }
+    }
+
+    private void createDepProposal(
+        IInvocationContext context, Collection<IJavaCompletionProposal> proposals, ServiceContainer bundle )
+    {
+        final String bundleName = bundle.getBundleName();
+        final String bundleVersion = bundle.getBundleVersion();
+
+        proposals.add(
+            new CUCorrectionProposal( "Add Liferay Dependence " + bundleName, context.getCompilationUnit(), null, -0 )
+            {
+
+                @Override
+                public void apply( IDocument document )
+                {
+                    try
+                    {
+                        GradleDependencyUpdater updater = new GradleDependencyUpdater( gradleFile.getLocation().toFile() );
+                        List<GradleDependency> existDependencies = updater.getAllDependencies();
+                        GradleDependency gd;
+
+                        if( bundleName.equals( "com.liferay.portal.kernel" ) )
```

Comment:
We need to add supprt for other dependencies that are very common like:
`org.osgi`, `javax.portlet`, and `javax.servlet`.  any dependency that is used in blade-samples or in project templates